### DISCUSSION TITLE
setQueryData `invalidate` option

### DISCUSSION THREAD

**Discussion by lekterable** [TanStack/query#1766] [Ideas] [answered]
I can see that 

`queryCache.setQueryData`
is very often used together with 
`queryCache.invalidateQueries`

for example:
```
queryCache.setQueryData('posts, [newPost]')
queryCache.invalidateQueries('posts')
```

what if we allowed `setQueryData` to take the third argument `options` and in particular, the option `invalidate`

if we had that we could simply do:
`queryCache.setQueryData('posts, [newPost]', { invalidate: true })`

which saves us a line of code on every occasion and provides a very useful abstraction

I'm totally down for working on a PR for that if you like it 🤙

**Comment by tannerlinsley** [comment, marked as answer]
I would instead urge users to create this abstraction utility on their own.  This will keep the size of the core library down

Tanner Linsley

**Reply by lekterable** [reply]
understood, thanks for the explanation and all your work! react-query is amazing :)

**Comment by TkDodo** [comment]
Invalidate will usually be done with fuzzy matching, while setQueryData is exact. If i have a list of tasks that i can filter, and i add a new task to the current view, I’ll call `setQueryData(['tasks', { filter }])` to optimistically add the task to the current view, but I’ll invalidate all task views with `invalidateQueries(['tasks'])`. 
so i don’t see the abstraction useful in many cases

**Reply by lekterable** [reply]
makes sense, thanks for the comprehensive answer!
